tfe/resource_tfe_variable.go

```diff
@@ -85,9 +84,25 @@ func resourceTFEVariable() *schema.Resource {
 			},
 		},
 
-		CustomizeDiff: customdiff.ForceNewIf("key", func(_ context.Context, d *schema.ResourceDiff, m interface{}) bool {
-			return d.Get("sensitive").(bool)
-		}),
+		CustomizeDiff: forceRecreateResourceIf(),
+	}
+}
+
+func forceRecreateResourceIf() schema.CustomizeDiffFunc {
+	/*
+		Destroy and add a new resource when:
+		1. the parameter key changed and the param sensitive is set to true
+		2. the parameter sensitive changed from true to false
+	*/
+	return func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+		wasSensitiveVar := d.HasChange("sensitive") && !(d.Get("sensitive").(bool))
+
+		if wasSensitiveVar {
+			d.ForceNew("sensitive")
```

Comment:
You should `return d.ForceNew("sensitive")` as that function returns an error and you want to capture that if it happens.

Reply:
Indeed!